src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/PropertyPageDataProducer.cs

```diff
@@ -116,8 +128,9 @@ public static IEntityValue CreatePropertyPageValue(IEntityRuntimeModel runtimeMo
             IEnumerable<IEntityValue> createPropertyPageValuesAsync()
             {
                 Rule? parentDebuggerPageRule = null;
-                
-                var propertyPageQueryContext = queryCacheProvider.CreateCache(project);
+
+                IPropertyPageQueryCache propertyPageQueryCache = queryCacheProvider.CreateCache(project);
+                QueryProjectPropertiesContext context = new QueryProjectPropertiesContext(isProjectFile: true, project.FullPath, itemType: null, itemName: null);
```

Comment:
>itemType: null, itemName: null [](start = 129, length = 30)

Would it make sense for  the constructor to automatically set these to null if isProjectFile is true?

Reply:
When there are expected (and especially required) relationships between constructor arguments, it can be clearer and safer to create factory methods.

Reply:
@tmeschter your comment on `QueryProjectPropertiesContext.ProjectFile` makes it sound like that instance is useful generally, even without the specific project file path. Why not use that instance here?

Reply:
At this point I don't remember why I didn't use the singleton. I'll have to go back and puzzle it out.